sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala

```diff
@@ -573,4 +573,37 @@ class CliSuite extends SparkFunSuite with BeforeAndAfterAll with Logging {
     // the date formatter for `java.sql.LocalDate` must output negative years with sign.
     runCliWithin(1.minute)("SELECT MAKE_DATE(-44, 3, 15);" -> "-0044-03-15")
   }
+
+  test("SPARK-33100: Ignore a semicolon inside a bracketed comment in spark-sql") {
+    runCliWithin(1.minute)("/* SELECT 'test';*/ SELECT 'test';" -> "test")
+    runCliWithin(1.minute)(";;/* SELECT 'test';*/ SELECT 'test';" -> "test")
+    runCliWithin(1.minute)("/* SELECT 'test';*/;; SELECT 'test';" -> "test")
+    runCliWithin(1.minute)("SELECT 'test'; -- SELECT 'test';" -> "")
+    runCliWithin(1.minute)("SELECT 'test'; /* SELECT 'test';*/" -> "")
+    runCliWithin(1.minute)("/*$meta chars{^\\;}*/ SELECT 'test';" -> "test")
+    runCliWithin(1.minute)("/*\nmulti-line\n*/ SELECT 'test';" -> "test")
+    runCliWithin(1.minute)("/*/* multi-level bracketed*/ SELECT 'test';" -> "test")
+
+    val testHintTablePath = Utils.createTempDir()
+
+    runCliWithin(3.minute)(
+      "CREATE TEMPORARY VIEW t1 AS SELECT * FROM VALUES(1, 2), (2, 3), (3, 4), (4, 5), (5, 6)" +
+        " as t1(key, val);"
+        -> "",
+      s"""CREATE TABLE testHint(key int, val int) USING hive
+        |LOCATION '${testHintTablePath.getAbsolutePath}';""".stripMargin
+        -> "",
+      "INSERT OVERWRITE TABLE testHint SELECT /*+ broadcast(t1) */ a.* from t1 a join t1 b" +
+        " on a.key=b.val;"
+        -> ""
+    )
+
+    val dataFiles = testHintTablePath.listFiles().filterNot{ file =>
+      file.getName.startsWith(".") || file.getName.startsWith("_")
+    }
+    assert(dataFiles.size == 1)
+
+    runCliWithin(1.minute)("DROP TABLE testHint;" -> "")
+    Utils.deleteRecursively(testHintTablePath)
```

Comment:
Could you create a separate test unit for this test, e.g., `test("SPARK-33100: XXX") {`?